src/transformers/models/swiftformer/modeling_swiftformer.py

```diff
@@ -0,0 +1,726 @@
+                layer_scale_init_value * torch.ones(dim).unsqueeze(-1).unsqueeze(-1), requires_grad=True
+            )
+
+    def forward(self, x):
+        x = self.local_representation(x)
+        B, C, H, W = x.shape
+        if self.use_layer_scale:
+            x = x + self.drop_path(
+                self.layer_scale_1
+                * self.attn(x.permute(0, 2, 3, 1).reshape(B, H * W, C)).reshape(B, H, W, C).permute(0, 3, 1, 2)
+            )
+            x = x + self.drop_path(self.layer_scale_2 * self.linear(x))
+
+        else:
+            x = x + self.drop_path(
+                self.attn(x.permute(0, 2, 3, 1).reshape(B, H * W, C)).reshape(B, H, W, C).permute(0, 3, 1, 2)
+            )
+            x = x + self.drop_path(self.linear(x))
+        return x
+
+
+def Stage(
+    dim,
+    index,
+    layers,
+    mlp_ratio=4.0,
+    act_layer=nn.GELU,
+    drop_path_rate=0.0,
+    use_layer_scale=True,
+    layer_scale_init_value=1e-5,
+    vit_num=1,
+):
+    """
+    Implementation of each SwiftFormer stages. Here, SwiftFormerEncoderBlk used as the last block in all stages, while
+    ConvEncoder used in the rest of the blocks. Input: tensor in shape [B, C, H, W] Output: tensor in shape [B, C, H,
+    W]
+    """
+    blocks = []
+
+    for block_idx in range(layers[index]):
+        block_dpr = drop_path_rate * (block_idx + sum(layers[:index])) / (sum(layers) - 1)
+
+        if layers[index] - block_idx <= vit_num:
+            blocks.append(
+                SwiftFormerEncoderBlk(
+                    dim,
+                    mlp_ratio=mlp_ratio,
+                    act_layer=act_layer,
+                    drop_path=block_dpr,
+                    use_layer_scale=use_layer_scale,
+                    layer_scale_init_value=layer_scale_init_value,
+                )
+            )
+
+        else:
+            blocks.append(ConvEncoder(dim=dim, hidden_dim=int(mlp_ratio * dim), kernel_size=3))
+
+    blocks = nn.Sequential(*blocks)
+    return blocks
+
+
+class SwiftFormerEncoder(nn.Module):
+    def __init__(self, config: SwiftFormerConfig) -> None:
+        super().__init__()
+        self.config = config
+
+        embed_dims = config.embed_dims
+        mlp_ratios = config.mlp_ratios
+        downsamples = config.downsamples
+        act_layer = ACT2CLS[config.act_layer]
+
+        down_patch_size = config.down_patch_size
+        down_stride = config.down_stride
+        down_pad = config.down_pad
+        drop_path_rate = config.drop_path_rate
+        use_layer_scale = config.use_layer_scale
+        layer_scale_init_value = config.layer_scale_init_value
+        vit_num = config.vit_num
+        layers = config.layers
+
+        # Transformer model
+        network = []
+        for i in range(len(layers)):
+            stage = Stage(
+                embed_dims[i],
+                i,
+                layers,
+                mlp_ratio=mlp_ratios,
+                act_layer=act_layer,
```

Comment:
Rather than instantiating and passing the `act_layer` above, the stage layers should take the `config` onject and create the activation layer using `config,act_layer`